service/api/supervisorhttp/http.go

```diff
@@ -0,0 +1,146 @@
+/*
+Supervisorhttp provides the HTTP API for working with Supervisor.
+*/
+package supervisorhttp
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/LeonidVas/tarantool-supervisor/service/supervisor"
+)
+
+// SupervisorHandler is used to communicate with the Supervisor over HTTP.
+type SupervisorHandler struct {
+	sv *supervisor.Supervisor
+}
+
+// params structure contains all the command parameters
+// that can be passed through the HTTP API.
+type params struct {
+	// Inst - Instance name.
+	Inst string `json:"inst"`
+	// Env - environment variables for the starting Instance.
+	Env []string `json:"env"`
+	// IsRestart - the setting is responsible for the
+	// need to restart the Instance on failure.
+	// Default: true.
+	IsRestart bool `json:"is_restart"`
+	// id - Instance ID.
+	Id int `json:"id"`
+	// Force - the setting is responsible for "force" termination
+	// the Instance in case of a graceful termination failure.
+	// Default: true.
+	Force bool `json:"force"`
+}
```

Comment:
> There are both `Inst` and `Id`. What about `InstanceName` and `InstanceID`? Or `Name` and `ID`? Moreover, this two params can be placed one-by-one to simplify understanding the code.
> 

I renamed `Inst` and `Id` to `Name` and `ID`.
But I don't understand, do you suggest any other changes?

> Some parameters are used for one command, would you like to use prefixes? Smth like
> 
> `Force` -> `StopForce`
> `Env` -> `StartEnv`
> `IsRestart` -> `StartEnable`

I don't think the prefix is needed here. I don't want to associate the parameter name with the command in which this parameter will be used. The same parameter can be reused by another command in the future. `IsRestart` - IMHO is easier to understand than `Enable`. In addition, `Enable` assumes automatic start after reboot, and this is outside the area of responsibility of `IsRestart`.

Reply:
But `Force` without `Stop` prefix doesn't make sense. I understand that `Force` doesn't relate to `list` or `status`, but it confuses now. 

You are right about `enabled`, I suggest renaming `IsRestart` to `Restart` (because it's a verb).

IMHO, it's better to have separated `<command>Params` or `<command>Opts` for each command. `command.Params` can be of type `interface{}` and later converted to specific command opts. Current implementation doesn't provide full request format check.

Reply:
Updated parameter check. Strict mode is now used and method invocation has been decoupled from parameter validation.

Reply:
Yep. But I look at it as a failed restart attempt (for a function). The reason will be correctly displayed in the error and the event will be saved to the log.